### ISSUE TITLE
[turborepo] turbo prune error

### ISSUE THREAD

**Issue by ThomasMarcelo** [vercel/turbo#5595]
### What version of Turborepo are you using?

1.10.10

### What package manager are you using / does the bug impact?

Yarn v1, Yarn v2/v3 (node_modules linker only)

### What operating system are you using?

Linux

### Describe the Bug

The `turbo prune --scope=<app> --docker` command throws this error during a Docker image build:
```
thread 'main' panicked at 'package.json path should have parent', crates/turborepo-lib/src/package_graph/builder.rs:459:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### Expected Behavior

It should prune the application with a message like:
```
Generating pruned monorepo for <app> in /out
[...]
```

### To Reproduce

Add `RUN turbo prune --scope=<app> --docker` to a Dockerfile and run the docker build command.

**Comment by chris-olszewski** [comment]
Hi @ThomasMarcelo, if possible could you provide a reproduction repo? If you're unable to isolate the problem, could you provide:
 - What directory are you running prune from
 - The output of `turbo info`

**Comment by tannerbaum** [comment]
@ThomasMarcelo I hit this briefly and realised that I was missing a WORKDIR invocation in my Dockerfile. Could this be the case for you?

**Comment by ThomasMarcelo** [comment]
> @ThomasMarcelo I hit this briefly and realized I was missing a WORKDIR invocation in my Dockerfile. Could this be the case for you?

I fixed our version to 1.10.09, which worked before.
